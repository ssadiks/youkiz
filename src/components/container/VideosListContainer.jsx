import { connect } from 'react-redux';
import VideosList from '../presentational/VideosList/VideosList';
import { fetchVideosAction, fetchDancersAction } from '../../redux/actions';

const mapStateToProps = state => ({
  videosList: state.videosReducer.videosList,
  dancersList: state.dancersReducer.dancersList,
  isPending: state.videosReducer.isPending,
  error: state.videosReducer.error,
});

const VideosListContainer = connect(mapStateToProps, {
  fetchVideosAction,
  fetchDancersAction
})(VideosList);

export default VideosListContainer;
