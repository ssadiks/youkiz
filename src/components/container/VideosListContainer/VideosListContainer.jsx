import { connect } from 'react-redux';
import VideosList from '../../presentational/VideosList/VideosList';
import {
  fetchVideosAction,
  fetchDancersAction,
  deleteVideoAction,
  openModalDialog,
  hideModalDialog,
  resetVideosListAction,
  toggleLazyLoading,
  updateVideosParams,
  resetVideosParamsAction
} from '../../../redux/actions/index';

const mapStateToProps = state => ({
  videosList: state.videosReducer.videosList,
  dancersList: state.dancersReducer.dancersList,
  isPending: state.videosReducer.isPending,
  error: state.videosReducer.error,
  params: state.lazyLoadingReducer.params
});

const VideosListContainer = connect(mapStateToProps, {
  fetchVideosAction,
  fetchDancersAction,
  deleteVideoAction,
  openModalDialog,
  hideModalDialog,
  resetVideosListAction,
  toggleLazyLoading,
  updateVideosParams,
  resetVideosParamsAction
})(VideosList);

const loadData = (store) => {
  return store.dispatch(fetchVideosAction());
}

export default {
  loadData,
  component: VideosListContainer
};
