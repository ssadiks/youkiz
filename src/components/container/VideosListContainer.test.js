import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import VideosListContainer from './VideosListContainer';

configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const INITIAL_STATE = {
  videosReducer: {
    videosList: [],
    isPending: false,
    error: null
  },
  dancersReducer: {
    dancersList: [],
  }
};

describe('VideosListContainer', () => {
  it('renders with default props', () => {
    const store = mockStore(INITIAL_STATE);
    const wrapper = shallow(
      <VideosListContainer store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
