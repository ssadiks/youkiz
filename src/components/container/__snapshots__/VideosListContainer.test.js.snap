// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosListContainer renders with default props 1`] = `
<VideosList
  dancersList={Array []}
  deleteVideoAction={[Function]}
  error={null}
  fetchDancersAction={[Function]}
  fetchVideosAction={[Function]}
  isPending={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userConnected={false}
  videosList={Array []}
/>
`;
